doc/manual/src/contributing/experimental-flags.md

```diff
@@ -0,0 +1,68 @@
+This section describes the notion of *experimental features*, and how it fits into the big picture of the development of Nix.
+
+# What are experimental features?
+
+[Since Nix 2.4](../release-notes/rl-2.4.md), some features of Nix can be flagged as experimental.
+Experimental features can be changed or removed at any time, and users must [explicitly enable them].
+This allows accessing new functionality that is not yet stable without unwittingly relying on them.
+
+[explicitly enable them]: ../nix-conf.md#conf-expeirmental-features
+
+# When should a new feature be marked experimental?
+
+A change in the Nix codebase should be guarded by an experimental feature flag if it is likely to either introduce a regression of any kind, or if it implies a non-trivial change to the external interface of Nix.
```

Comment:
I would drop the "introduce a regression of any kind". Any change can introduce a regression, but that's a matter of testing.

Reply:
Is it the intent that non-trivial API changes will be rolled out first as experimental features?

Reply:
```suggestion
A change in the Nix codebase should be guarded by an experimental feature flag if it implies a backwards-incompatible or non-trivial change to the external interface of Nix.
```